src/renderer/backend/opengl3.rs

```diff
@@ -375,9 +392,52 @@ impl TextureHeap {
         );
         gl.BindTexture(gl::TEXTURE_2D, bound_texture as _);
 
-        self.textures.push(texture);
+        let id = TextureId::from(self.textures.len());
+        self.textures.push(Texture { gl_texture: texture, width, height });
+
+        Ok(id)
+    }
+
+    unsafe fn update_texture(
+        &mut self,
+        gl: &gl::Gl,
+        texture: TextureId,
+        data: &[u8],
+        width: u32,
+        height: u32,
+    ) -> Result<()> {
+        let texture_info = self.get(texture);
+        if texture_info.width != width || texture_info.height != height {
+            return Result::Err(Error::new(
```

Comment:
Same observation as above, just marking the spot with a  comment for convenience.

(is the `Result::` prefix necessary here?)